support_bundle/get_support_bundle.sh

```diff
@@ -11,216 +11,367 @@ catch() {
+            fi
+        done
+
+        get_agent_version_metric_limits
+    fi
+
+    # Configure kubectl command if labels are set
+    if [[ -z ${LABELS} ]]; then
+        SYSDIGCLOUD_PODS=$(kubectl ${KUBE_OPTS} get pods | awk '{ print $1 }' | grep -v NAME)
+    else
+        SYSDIGCLOUD_PODS=$(kubectl ${KUBE_OPTS} -l "role in (${LABELS})" get pods | awk '{ print $1 }' | grep -v NAME)
+    fi
+
+    echo "Using namespace ${NAMESPACE}"
+    echo "Using context ${CONTEXT}"
+
+    # Collect container logs for each pod
+    if [[ "${SKIP_LOGS}" == "false" ]]; then
+        echo "Gathering Logs from ${NAMESPACE} pods"
+        command='tar czf - /logs/ /opt/draios/ /var/log/sysdigcloud/ /var/log/cassandra/ /tmp/redis.log /var/log/redis-server/redis.log /var/log/mysql/error.log /opt/prod.conf 2>/dev/null || true'
+        for pod in ${SYSDIGCLOUD_PODS};
+        do
+            echo "Getting support logs for ${pod}"
+            mkdir -p ${LOG_DIR}/${pod}
+            containers=$(kubectl ${KUBE_OPTS} get pod ${pod} -o json | jq -r '.spec.containers[].name')
+            for container in ${containers}; do
+                kubectl ${KUBE_OPTS} logs ${pod} -c ${container} ${SINCE_OPTS} > ${LOG_DIR}/${pod}/${container}-kubectl-logs.txt
+                echo "Execing into ${container}"
+                kubectl ${KUBE_OPTS} exec ${pod} -c ${container} -- bash >/dev/null 2>&1 && RETVAL=$? || RETVAL=$? && true
+                kubectl ${KUBE_OPTS} exec ${pod} -c ${container} -- sh >/dev/null 2>&1 && RETVAL1=$? || RETVAL1=$? && true
+                if [ $RETVAL -eq 0 ]; then
+                    kubectl ${KUBE_OPTS} exec ${pod} -c ${container} -- bash -c "${command}" > ${LOG_DIR}/${pod}/${container}-support-files.tgz || true
+                elif [ $RETVAL1 -eq 0 ]; then
+                    kubectl ${KUBE_OPTS} exec ${pod} -c ${container} -- sh -c "${command}" > ${LOG_DIR}/${pod}/${container}-support-files.tgz || true
+                else
+                    echo "Skipping log gathering for ${pod}"
+                fi
+            done
+        done
+    fi
+
+    echo "Gathering pod descriptions"
+    for pod in ${SYSDIGCLOUD_PODS}; do
+        echo "Getting pod description for ${pod}"
+        mkdir -p ${LOG_DIR}/${pod}
+        kubectl ${KUBE_OPTS} get pod ${pod} -o json > ${LOG_DIR}/${pod}/kubectl-describe.json
+    done
+
+    #Collect Describe Node Output
+    echo "Collecting node information"
+    kubectl ${KUBE_OPTS} describe nodes | tee -a ${LOG_DIR}/describe_node_output.log || echo "No permission to describe nodes!"
+
+    NODES=$(kubectl ${KUBE_OPTS} get nodes --no-headers | awk '{print $1}') && RETVAL=0 || { RETVAL=$? && echo "No permission to get nodes!"; }
+    if [[ "${RETVAL}" == "0" ]]; then
+        mkdir -p ${LOG_DIR}/nodes
+        for node in ${NODES[@]}; do
+            kubectl ${KUBE_OPTS} get node ${node} -ojson > ${LOG_DIR}/nodes/${node}-kubectl.json
+        done
+        unset RETVAL
```

Comment:
should this be outside of the `if` block?

Reply:
Yes, if they choose to skip logs, we still want to get the json description of the pod (small amount of data) and allows us to still analyze the if needed.  That's why I moved it outside of the log block, as technically it isn't a "log"